src/qrules/particle.py

```diff
@@ -221,22 +241,29 @@ def _repr_pretty_(self, p: PrettyPrinter, cycle: bool) -> None:
                         p.breakable()
                         p.text(f"{attribute.name}=")
                         if isinstance(value, Parity):
-                            p.text(_to_fraction(int(value), render_plus=True))
+                            p.text(_int_as_signed_str(int(value), render_plus=True))
                         else:
                             p.pretty(value)  # type: ignore[attr-defined]
                         p.text(",")
             p.breakable()
             p.text(")")
 
 
+def _int_as_signed_str(value: int, render_plus: bool = False) -> str:
```

Comment:
Functions like these occur quite frequently in the code base. Can they be combined somehow?

Reply:
You mean with a wrapper function wich then "switches" on the types (the Python analogue)?

Reply:
Using https://github.com/ComPWA/qrules/pull/288#discussion_r1846949473, you can render any `SupportsFloat` object (I think) using the same implementation.

Reply:
Some renderers/to_str-converters were combined, although there is still some redundancy for rendering `Fraction`s, because fusing those would result in circular dependencies.